library/ix-dev/community/overseerr/templates/__overseerr.tpl

```diff
@@ -0,0 +1,95 @@
+{{- define "overseerr.workload" -}}
+workload:
+  overseerr:
+    enabled: true
+    primary: true
+    type: Deployment
+    podSpec:
+      hostNetwork: {{ .Values.overseerrNetwork.hostNetwork }}
+      containers:
+        overseerr:
+          enabled: true
+          primary: true
+          imageSelector: image
+          securityContext:
+            runAsUser: {{ .Values.overseerrRunAs.user }}
+            runAsGroup: {{ .Values.overseerrRunAs.group }}
+          env:
+            PORT: {{ .Values.overseerrNetwork.webPort }}
+          {{ with .Values.overseerrConfig.additionalEnvs }}
+            {{ range $env := . }}
+            {{ $env.name }}: {{ $env.value }}
+            {{ end }}
+          {{ end }}
+          probes:
+            liveness:
+              enabled: true
+              type: http
+              port: {{ .Values.overseerrNetwork.webPort }}
+              path: /ping
+            readiness:
+              enabled: true
+              type: http
+              port: {{ .Values.overseerrNetwork.webPort }}
+              path: /ping
+            startup:
+              enabled: true
+              type: http
+              port: {{ .Values.overseerrNetwork.webPort }}
+              path: /ping
+      initContainers:
+      {{- include "ix.v1.common.app.permissions" (dict "containerName" "01-permissions"
+                                                        "UID" .Values.overseerrRunAs.user
+                                                        "GID" .Values.overseerrRunAs.group
+                                                        "mode" "check"
+                                                        "type" "init") | nindent 8 }}
+
+{{/* Service */}}
+service:
+  overseerr:
+    enabled: true
+    primary: true
+    type: NodePort
+    targetSelector: overseerr
+    ports:
+      webui:
+        enabled: true
+        primary: true
+        port: {{ .Values.overseerrNetwork.webPort }}
+        nodePort: {{ .Values.overseerrNetwork.webPort }}
+        targetSelector: overseerr
+
+{{/* Persistence */}}
+persistence:
+  config:
+    enabled: true
+    type: {{ .Values.overseerrStorage.config.type }}
+    datasetName: {{ .Values.overseerrStorage.config.datasetName | default "" }}
+    hostPath: {{ .Values.overseerrStorage.config.hostPath | default "" }}
+    targetSelector:
+      overseerr:
+        overseerr:
+          mountPath: /app/config
+        01-permissions:
+          mountPath: /mnt/directories/config
+  tmp:
+    enabled: true
+    type: emptyDir
+    targetSelector:
+      overseerr:
+        overseerr:
+          mountPath: /tmp
```

Comment:
Let's remove it for now, unless the app needs it

Reply:
I believe the `/tmp` directory is needed for Yarn to install the required NodeJS packages, see [this relevant issue to yarn](https://github.com/yarnpkg/yarn/issues/6685).

Reply:
I don't think overseerr will install packages at runtime, containers are meant to be ready to use (with all the deps inside)

Yep, overseerr install those packages at build time (before publishing the container)
https://github.com/sct/overseerr/blob/a4d07f5afab613317d96c9c6e9b47157a5a28986/Dockerfile#L27

Reply:
It looks like Yarn is actually needed at runtime in the container for testing I removed the `/tmp` mount and yarn started throwing errors and the server won't start,

```
2023-05-02 01:38:24.982171+00:00yarn run v1.22.19
2023-05-02 01:38:25.019210+00:00warning Skipping preferred cache folder "/.cache/yarn" because it is not writable.
2023-05-02 01:38:25.019243+00:00warning Skipping preferred cache folder "/tmp/.yarn-cache-568" because it is not writable.
2023-05-02 01:38:25.019273+00:00warning Skipping preferred cache folder "/tmp/.yarn-cache" because it is not writable.
2023-05-02 01:38:25.029535+00:00error Yarn hasn't been able to find a cache folder it can use. Please use the explicit --cache-folder option to tell it what location to use, or make one of the preferred locations writable.
2023-05-02 01:38:25.029623+00:00info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

```

When the `/tmp` path is added I can see yarn being using

```
2023-05-02 01:41:47.541134+00:00yarn run v1.22.19
2023-05-02 01:41:47.578176+00:00warning Skipping preferred cache folder "/.cache/yarn" because it is not writable.
2023-05-02 01:41:47.578201+00:00warning Selected the next writable cache folder in the list, will be "/tmp/.yarn-cache-568".
2023-05-02 01:41:47.585816+00:00$ NODE_ENV=production node dist/index.js
2023-05-02 01:41:47.590341+00:00warning Cannot find a suitable global folder. Tried these: "/usr/local, /.yarn"
2023-05-02 01:41:49.162592+00:002023-05-02T01:41:49.161Z [[32minfo[39m]: Commit Tag: $GIT_SHA 
2023-05-02 01:41:49.523775+00:002023-05-02T01:41:49.523Z [[32minfo[39m]: Starting Overseerr version 1.32.5 
2023-05-02 01:41:49.870552+00:00warn  - You have enabled experimental feature (scrollRestoration) in next.config.js.
2023-05-02 01:41:49.870612+00:00warn  - Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use at your own risk.
2023-05-02 01:41:49.870619+00:002023-05-02T01:41:49.870619374Z
2023-05-02 01:41:50.857885+00:002023-05-02T01:41:50.857Z [[32minfo[39m][Notifications]: Registered notification agents 
2023-05-02 01:41:50.890822+00:002023-05-02T01:41:50.890Z [[32minfo[39m][Jobs]: Scheduled jobs loaded 
2023-05-02 01:41:50.896522+00:002023-05-02T01:41:50.896Z [[32minfo[39m][Discover Slider]: Creating built-in discovery slider {"slider":{"type":1,"enabled":true,"isBuiltIn":true,"order":0}}
2023-05-02 01:41:50.930906+00:002023-05-02T01:41:50.930Z [[32minfo[39m][Discover Slider]: Creating built-in discovery slider {"slider":{"type":2,"enabled":true,"isBuiltIn":true,"order":1}}
2023-05-02 01:41:50.968678+00:002023-05-02T01:41:50.968Z [[32minfo[39m][Discover Slider]: Creating built-in discovery slider {"slider":{"type":3,"enabled":true,"isBuiltIn":true,"order":2}}
2023-05-02 01:41:51.007981+00:002023-05-02T01:41:51.007Z [[32minfo[39m][Discover Slider]: Creating built-in discovery slider {"slider":{"type":4,"enabled":true,"isBuiltIn":true,"order":3}}
2023-05-02 01:41:51.041967+00:002023-05-02T01:41:51.041Z [[32minfo[39m][Discover Slider]: Creating built-in discovery slider {"slider":{"type":5,"enabled":true,"isBuiltIn":true,"order":4}}
2023-05-02 01:41:51.071768+00:002023-05-02T01:41:51.071Z [[32minfo[39m][Discover Slider]: Creating built-in discovery slider {"slider":{"type":6,"enabled":true,"isBuiltIn":true,"order":5}}
2023-05-02 01:41:51.087170+00:002023-05-02T01:41:51.086Z [[32minfo[39m][Discover Slider]: Creating built-in discovery slider {"slider":{"type":7,"enabled":true,"isBuiltIn":true,"order":6}}
2023-05-02 01:41:51.096427+00:002023-05-02T01:41:51.096Z [[32minfo[39m][Discover Slider]: Creating built-in discovery slider {"slider":{"type":8,"enabled":true,"isBuiltIn":true,"order":7}}
2023-05-02 01:41:51.123240+00:002023-05-02T01:41:51.123Z [[32minfo[39m][Discover Slider]: Creating built-in discovery slider {"slider":{"type":9,"enabled":true,"isBuiltIn":true,"order":8}}
2023-05-02 01:41:51.148457+00:002023-05-02T01:41:51.148Z [[32minfo[39m][Discover Slider]: Creating built-in discovery slider {"slider":{"type":10,"enabled":true,"isBuiltIn":true,"order":9}}
2023-05-02 01:41:51.174715+00:002023-05-02T01:41:51.174Z [[32minfo[39m][Discover Slider]: Creating built-in discovery slider {"slider":{"type":11,"enabled":true,"isBuiltIn":true,"order":10}}
2023-05-02 01:41:51.200372+00:002023-05-02T01:41:51.200Z [[32minfo[39m][Discover Slider]: Creating built-in discovery slider {"slider":{"type":12,"enabled":true,"isBuiltIn":true,"order":11}}
2023-05-02 01:41:51.400930+00:002023-05-02T01:41:51.400Z [[32minfo[39m][Server]: Server ready on port 30000 
2023-05-02 01:42:00.008468+00:002023-05-02T01:42:00.008Z [[34mdebug[39m][Jobs]: Starting scheduled job: Download Sync 

```

And the contents of /tmp in the container are here,

```
/app $ ls /tmp/
yarn--1682991707590-0.23918655606615036
/app $ 
```

Reply:
Cool, thanks for checking.